pkg/scheduler/core/generic_scheduler.go

```diff
@@ -278,6 +279,32 @@ func (g *genericScheduler) findNodesThatFitPod(ctx context.Context, prof *profil
 
 	}
 
+	// The "NominatedNodeName" field is set due to the preemption in the previous scheduling cycle, the node
+	// most likely is still the only candidate for the preemptor pod.
+	// Give a try on this node directly when the option preemptorOnNominatedNode is set.
+	if g.preemptorOnNominatedNode && len(pod.Status.NominatedNodeName) > 0 {
+		nnn := pod.Status.NominatedNodeName
+		nodeInfo, err := g.nodeInfoSnapshot.Get(nnn)
+		if err != nil {
+			return nil, nil, err
+		}
+		fits, status, err := PodPassesFiltersOnNode(ctx, prof.PreemptHandle(), state, pod, nodeInfo)
+		var feasibleNodes []*v1.Node
+		if fits {
+			feasibleNodes = []*v1.Node{nodeInfo.Node()}
+		} else {
+			filteredNodesStatuses[nodeInfo.Node().Name] = status
+			// As only the nominated node is validated this time, it's possible there is node in the cluster is already
+			// available for this pod, return the error so that a fresh new scheduling cycle is expected later on.
+			err = fmt.Errorf("Nominated node is not a good candidate anymore, error: %v", err.Error())
+		}
+		// Cleanup the nominated node from the pod so that the node won't affect the future scheduling cycle.
+		if errs := util.ClearNominatedNodeName(prof.ClientSet(), pod); errs != nil {
+			klog.Errorf("Cannot clear 'NominatedNodeName' field of pod %v/%v: %v", pod.Namespace, pod.Name, errs)
+		}
+		return feasibleNodes, filteredNodesStatuses, err
```

Comment:
a question about do we want to bypass the extender which invoking at line 313?

also there will bypassing some logic inside 'findNodesThatPassFilters. another way, if we call the 'findNodesThatPassFilters' and add new parameter to it about the nodes we want to check, instead of the hardcode all nodes at line 322, then it won't be a problem.

Reply:
Done